pennylane/transforms/condition.py

```diff
@@ -93,8 +102,72 @@ def qnode():
             return qml.expval(qml.PauliZ(1))
     """
 
-    @wraps(then_op)
-    def wrapper(*args, **kwargs):
-        return Conditional(measurement, then_op(*args, do_queue=False, **kwargs))
+    if inspect.isclass(then_func) and issubclass(then_func, Operator):
+
+        @wraps(then_func)
+        def wrapper(*args, **kwargs):
+            ops = []
+            then_cond_op = Conditional(measurement, then_func(*args, do_queue=False, **kwargs))
+            ops.append(then_cond_op)
+
+            if else_func:
+
+                # Copy the MV such that the external state is not changed
+                inverted_m = copy(measurement)
+                else_cond_op = Conditional(~inverted_m, else_func(*args, do_queue=False, **kwargs))
+                ops.append(else_cond_op)
+
+            return ops

```

Comment:
@antalszava on retrospect, I'm not sure if this branch here is needed? It looks like the `else` branch below should work for all use cases?

Reply:
If you consider `then_func = qml.RX`, then `then_func` is still a quantum function, just with a single operation. So 

```python
tape = make_tape(then_func)(*args, **kwargs)

for op in tape.operations:
    Conditional(measurement, op)
```

should work perfectly fine (in this example, `tape.operations == [qml.RX(..)]`).

Reply:
If this is right, I would be in favour of just using the `else` branch below - avoiding having two branches and simplifying the logic to one pathway could reduce the likelihood of bugs.

Reply:
Oh right! Great point! It works, removed in https://github.com/PennyLaneAI/pennylane/pull/2275/commits/9eec1c208ff192cd8c0ab35f3ae645d830e78e6d.